prov/gni/src/gnix_sep.c

```diff
@@ -886,6 +886,146 @@ gnix_sep_msg_injectdata(struct fid_ep *ep, const void *buf, size_t len,
 			  NULL, flags, data, 0);
 }
 
+DIRECT_FN STATIC ssize_t gnix_sep_trecv(struct fid_ep *ep, void *buf,
+					size_t len,
+					void *desc, fi_addr_t src_addr,
+					uint64_t tag, uint64_t ignore,
+					void *context)
+{
+	struct gnix_fid_trx *rx_ep = container_of(ep, struct gnix_fid_trx,
+						  ep_fid);
+
+	return _ep_recv(&rx_ep->ep->ep_fid, buf, len, desc, src_addr, context,
+			FI_TAGGED, tag, ignore);
+}
+
+DIRECT_FN STATIC ssize_t gnix_sep_trecvv(struct fid_ep *ep,
+					 const struct iovec *iov,
+					 void **desc, size_t count,
+					 fi_addr_t src_addr,
+					 uint64_t tag, uint64_t ignore,
+					 void *context)
+{
+	struct gnix_fid_trx *rx_ep = container_of(ep, struct gnix_fid_trx,
+						  ep_fid);
+
+	return _ep_recvv(&rx_ep->ep->ep_fid, iov, desc, count, src_addr,
+			 context, FI_TAGGED, tag, ignore);
+}
+
+DIRECT_FN STATIC ssize_t gnix_sep_trecvmsg(struct fid_ep *ep,
+					   const struct fi_msg_tagged *msg,
+					   uint64_t flags)
+{
+	uint64_t clean_flags;
+
+	const struct fi_msg _msg = {
+			.msg_iov = msg->msg_iov,
+			.desc = msg->desc,
+			.iov_count = msg->iov_count,
+			.addr = msg->addr,
+			.context = msg->context,
+			.data = msg->data
+	};
+
+	clean_flags = (flags & (GNIX_TRECVMSG_FLAGS)) | FI_TAGGED;
```

Comment:
I'm tending to agree with @jswaro but note that in commit:

commit a40353f5b015df18d30ca27ff5c111903b1a897d
Author: James Swaro <jswaro@cray.com>
Date:   Thu Sep 24 13:55:15 2015 -0500
    implement FI_PEEK|CLAIM|DISCARD

Jim was the last one to touch this code so this is third time we're reviewing it.